ksqldb-common/src/main/java/io/confluent/ksql/function/AggregateFunctionFactory.java

```diff
@@ -73,8 +73,7 @@ public AggregateFunctionFactory(final UdfMetadata metadata) {
     this.metadata = Objects.requireNonNull(metadata, "metadata can't be null");
   }
 
-  public abstract KsqlAggregateFunction<?, ?, ?> createAggregateFunction(
-      List<SqlArgument> argTypeList, AggregateFunctionInitArguments initArgs);
+  public abstract FunctionSource getFunction(List<SqlType> argTypeList);
```

Comment:
Zara and I discussed this on Slack, and this is not a regression. In the old implementation and the current implementation, initial args must be `Literal`s, but time units are not `Literal`s (`IntervalUnit` extends `Expression` instead of `Literal`). This might be reported as a separate issue.